webpack/assets/javascripts/react_app/Root/Context/ForemanContext.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+const ForemanContext = React.createContext({});
```

Comment:
Can we set the app metadata as the default value and then any component that doesn't have a provider in the tree will get it for free? i.e. instead of wrapping all of the components mounted from ERB. 
from the react docs:

> The defaultValue argument is only used when a component does not have a matching Provider above it in the tree.  This can be helpful for testing components in isolation without wrapping them.

Reply:
sounds interesting, though `ForemanContext` is created once before any rendering, and `app metadata` arrives from the server during the rendering. we can find a workaround for it, but doesn't it sound like an anti-pattern? react document mentions that it may be useful for testing, not for populate data for global use without a provider above the tree.